internal/cli/server_upgrade.go

```diff
@@ -446,6 +448,38 @@ func (c *ServerUpgradeCommand) upgradeRunner(
 					runnerDefaultNames = append(runnerDefaultNames, fmt.Sprintf(runnerDefaultName, cfg.Name))
 					runnerUnsetStr = append(runnerUnsetStr, fmt.Sprintf(runnerUnsetDefault, cfg.PluginType, cfg.Name))
 				}
+
+				// Checking if the default runner profile for a platform, pre-0.9, has the correct task launcher configs
+				if c.platform == cfg.Name {
+					switch c.platform {
+					case "kubernetes":
+						// attempt to parse the runner profile config into the correct task launcher config struct
+						var result *k8s.TaskLauncherConfig
+						err = json.Unmarshal(cfg.PluginConfig, result)
```

Comment:
I think we also have to check if `cfg.ConfigFormat == pb.Hcl_JSON` in case their default runner profile was saved as HCL instead of json.